en/docs/administer/changing-to-ibm-db2.md

```diff
@@ -1,84 +1,155 @@
+      <td>The hostname of the host where database is hosted.</td>
+      </tr>
+      <tr class="even">
+      <td><strong>port</strong></td>
+      <td>The port of the database.</td>
+      </tr>
+      <tr class="even">
+      <td><strong>name</strong></td>
+      <td>The name of the database.</td>
+      </tr>
+      </table>   
+ 
+ A Sample configuration is given below.
+
+   1. `WSO2_IDENTITY_DB` 
+    
+       1. `deployment.toml` Configurations
+
+           ``` toml
+           [database.identity_db]
+           type = "db2"
+           hostname = "localhost"
+           name = "regdb"
+           username = "regadmin"
+           password = "regadmin"
+           port = "50000"
+           ```
+       
+       1. Executing database scripts.
+        
+          Navigate to `<IS-HOME>/dbscripts`. Execute the scripts in the following files, against the database created.
+           
+           - `<IS-HOME>/dbscripts/identity/db2.sql`
+           - `<IS-HOME>/dbscripts/identity/uma/db2.sql`
+           - `<IS-HOME>/dbscripts/consent/db2.sql`
+         
+   2. `WSO2_SHARED_DB`
+        
+       1. `deployment.toml` Configurations
+
+           ``` toml
+           [database.shared_db]
+           type = "mssql"
+           hostname = "localhost"
+           name = "regdb"
+           username = "regadmin"
+           password = "regadmin"
+           port = "50000"
+           ```
+           
+       1. Executing database scripts.
+        
+          Navigate to `<IS-HOME>/dbscripts`. Execute the scripts in the following file, against the database created.
+                      
+           - `<IS-HOME>/dbscripts/db2.sql`
+           
+   3. If you have a requirement in using workflow feature follow, 
+       [Changing the default database of BPS database](../../administer/changing-datasource-bpsds)
+       
+   4.  Download the DB2 JDBC driver for the version you are using and
+            copy it to the `<IS_HOME>/repository/components/lib` folder  
+    
+    !!! note     
+        In earlier versions WSO2 Identity Server had the option to create databases automatically using the 
+        -DSetup option  **from [January 2018 onwards](https://wso2.com/products/carbon/release-matrix/) 
+        WSO2 Identity Server has deprecated the** **`              -DSetup             `** **option**
+        Note that the proper practice is for the DBA to run the DDL statements manually so that the DBA
+        can examine and optimize any DDL statement (if necessary) based on the DBA best practices that are in
+        place within the organization.  
+           
+            
 
-Follow the steps below to change the type of the default datasource.
+   2.**Advanced Database Configurations.**
 
-1.  Edit the default datasourceconfigurationin the \<
-    `            IS_HOME>/repository/conf/deployment.toml file as shown below.
+Apart from above basic configurations WSO2 Identity Server supports advanced database configurations.
```

Comment:
Apart from the basic configurations specified above, WSO2 Identity Server supports some advanced database configurations as well.

Reply:
done